# Exploración de datos PAES 2026

En este notebook realzaremos una primera revisión de los conjuntos de datos verificados en el paso `00_verificacion_entorno`. El objetivo es conocer la estructura y contenido en terminos de datos de cada fuente, utilizando como referencia el archivo PDF que acompaña a cada fuente y que entrega, entre otros, información sobre: observaciones según proceso de admisión, responsable del dato en particular, Variables (nombre, tipo, descripción y valores) asi como también códigos asociados a las unidades territoriales. 

Para esta experiencia se toma como referencia los contenidos de clases, evaluando en cada conjunto lo siguiente: 

- dimensiones del conjunto
- primeras observaciones
- variables y tipos de datos
- valores faltantes
- posibles duplicados
- variables cualitativas y cuantitativas relevantes
- posibles identificadores o llaves
- unidad de observación de cada tabla

El proceso se realizará con cada archivo, aun cuando esto podria hacerse de forma paralela. La idea es, apoyados en el PDF de cada archivo, ir generando un dominio acabado de cada dataset.  

La forma de trabajo definida utiliza los archivos disponibles en `data/raw` como referencia. Durante el proceso de análisis exploratorio y/o transformacion, los datasets quedaran en `data/intermediate`, en caso de que se requiera complementar o realizar algún paso adicional. Finalmente, el conjunto preparado y listo para análisis quedara en `data/processed`

In [24]:
from pathlib import Path
import pandas as pd

DIRECTORIO_ACTUAL = Path.cwd()
DIRECTORIO_RAIZ = DIRECTORIO_ACTUAL.parent
RAW_DIR = DIRECTORIO_RAIZ / "data" / "raw"


## Inscritos y Puntajes

Comenzaremos con el conjunto `A_INSCRITOS_PUNTAJES_PAES_2026_PUB_MRUN.csv`

In [25]:
RUTA_INSCRITOS = RAW_DIR / "PAES-2026-Inscritos-Puntajes" / "A_INSCRITOS_PUNTAJES_PAES_2026_PUB_MRUN.csv" #Definimos CARPETA / Archivo

# Primero miramos el archivo RAW. Si bien podriamos haber definido un comando para leer ya formateado el archivo, la idea es 
# poder entender en operación que esperar para definir un metodo de lectura. 
with open(RUTA_INSCRITOS, "r", encoding="utf-8") as archivo:
    for _ in range(3):
        print(archivo.readline())


MRUN;ANYO_PROCESO;COD_SEXO;FECHA_NACIMIENTO;RBD;CODIGO_ENS;LOCAL_EDUCACIONAL;UNIDAD_EDUCATIVA;NOMBRE_UNIDAD_EDUC;RAMA_EDUCACIONAL;DEPENDENCIA;CODIGO_REGION_EGRESO;NOMBRE_REGION_EGRESO;CODIGO_PROVINCIA_EGRESO;NOMBRE_PROVINCIA_EGRESO;CODIGO_COMUNA_EGRESO;NOMBRE_COMUNA_EGRESO;ANYO_DE_EGRESO;PROMEDIO_NOTAS;PTJE_NEM;PORC_SUP_NOTAS;PTJE_RANKING;CLEC_REG_ACTUAL;MATE1_REG_ACTUAL;MATE2_REG_ACTUAL;HCSOC_REG_ACTUAL;CIEN_REG_ACTUAL;CLEC_INV_ACTUAL;MATE1_INV_ACTUAL;MATE2_INV_ACTUAL;HCSOC_INV_ACTUAL;CIEN_INV_ACTUAL;CLEC_REG_ANTERIOR;MATE1_REG_ANTERIOR;MATE2_REG_ANTERIOR;HCSOC_REG_ANTERIOR;CIEN_REG_ANTERIOR;CLEC_INV_ANTERIOR;MATE1_INV_ANTERIOR;MATE2_INV_ANTERIOR;HCSOC_INV_ANTERIOR;CIEN_INV_ANTERIOR;PROMEDIO_CM_MAX;CLEC_MAX;MATE1_MAX;MATE2_MAX;HCSOC_MAX;CIEN_MAX;RINDIO_PROCESO_ANTERIOR;RINDIO_PROCESO_ACTUAL;HABILITACION_POST;PRUEBA_REG_CL;FORMA_REG_CL;CORRECTAS_REG_CL;ERRADAS_REG_CL;OMITIDAS_REG_CL;PRUEBA_REG_M1;FORMA_REG_M1;CORRECTAS_REG_M1;ERRADAS_REG_M1;OMITIDAS_REG_M1;PRUEBA_REG_M2;FORMA_REG_M2;CO

El resultado del codigo nos muestra que el separador utilizado en el .csv es `;` según el registro `MRUN;ANYO_PROCESO;COD_SEXO;FECHA_NACIMIENTO;RBD;`
En general uno podria asumir que para estos archivos .csv siempre el separador será `;` pero es una buena practica revisar el texto raw antes de declararlo en lectura como `viajes = pd.read_csv("viajes.csv", sep=";", decimal=",", low_memory=False)`


In [26]:
inscritos = pd.read_csv (RUTA_INSCRITOS, sep = ";", decimal= ",", low_memory=False)
inscritos

,MRUN,ANYO_PROCESO,COD_SEXO,FECHA_NACIMIENTO,RBD,CODIGO_ENS,LOCAL_EDUCACIONAL,UNIDAD_EDUCATIVA,NOMBRE_UNIDAD_EDUC,RAMA_EDUCACIONAL,...,PRUEBA_INV_CQUI,FORMA_INV_CQUI,CORRECTAS_INV_CQUI,ERRADAS_INV_CQUI,OMITIDAS_INV_CQUI,PRUEBA_INV_CTP,FORMA_INV_CTP,CORRECTAS_INV_CTP,ERRADAS_INV_CTP,OMITIDAS_INV_CTP
0,89,2026,2,200708,2308,310,1326,1,COLEGIO SAN ANTONIO DEL BALUARTE,H1,...,Ciencias Química.,0,0,0,0,Ciencias Técnico Profesional.,0,0,0,0
1,258,2026,1,200704,12305,310,2157,1,COLEGIO ALEXANDER FLEMING,H1,...,Ciencias Química.,0,0,0,0,Ciencias Técnico Profesional.,0,0,0,0
2,259,2026,2,200704,8005,510,2701,7,LICEO AYTUE,T2,...,Ciencias Química.,0,0,0,0,Ciencias Técnico Profesional.,0,0,0,0
3,496,2026,2,200711,10967,510,60,3,LICEO ELEUTERIO RAMIREZ MOLINA,T2,...,Ciencias Química.,0,0,0,0,Ciencias Técnico Profesional.,0,0,0,0
4,538,2026,2,200711,26538,310,4041,1,COLEGIO TERRAUSTRAL OESTE S.A,H1,...,Ciencias Química.,0,0,0,0,Ciencias Técnico Profesional.,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
320082,28540477,2026,1,200712,4708,410,1658,5,LICEO POLIVALENTE LAS SALINAS,T1,...,Ciencias Química.,0,0,0,0,Ciencias Técnico Profesional.,0,0,0,0
320083,28540478,2026,2,200705,218,610,61,4,LICEO FRANCISCO DE AGUIRRE,T3,...,Ciencias Química.,0,0,0,0,Ciencias Técnico Profesional.,0,0,0,0
320084,28540479,2026,2,200805,9900,610,836,5,COLEGIO BICENTENARIO DE EXCELENCIA POLITECNICO...,T3,...,Ciencias Química.,0,0,0,0,Ciencias Técnico Profesional.,0,0,0,0
320085,28540480,2026,1,200802,3110,310,6906,1,COMPLEJO EDUCACIONAL MAULE,H1,...,Ciencias Química.,0,0,0,0,Ciencias Técnico Profesional.,0,0,0,0


Con la tabla anterior de resultado, ya tenemos un dataframe conceptualmente reconocible, por lo que procedemos a realizar las exploraciones basicas revisadas en clases. 

In [27]:
inscritos.shape # Para conocer filas, columnas

(320087, 131)

In [28]:
inscritos.head() # Para conocer las primeras filas, aunque esto de alguna forma ya lo obtuvimos al mostrar inscritos

,MRUN,ANYO_PROCESO,COD_SEXO,FECHA_NACIMIENTO,RBD,CODIGO_ENS,LOCAL_EDUCACIONAL,UNIDAD_EDUCATIVA,NOMBRE_UNIDAD_EDUC,RAMA_EDUCACIONAL,...,PRUEBA_INV_CQUI,FORMA_INV_CQUI,CORRECTAS_INV_CQUI,ERRADAS_INV_CQUI,OMITIDAS_INV_CQUI,PRUEBA_INV_CTP,FORMA_INV_CTP,CORRECTAS_INV_CTP,ERRADAS_INV_CTP,OMITIDAS_INV_CTP
0,89,2026,2,200708,2308,310,1326,1,COLEGIO SAN ANTONIO DEL BALUARTE,H1,...,Ciencias Química.,0,0,0,0,Ciencias Técnico Profesional.,0,0,0,0
1,258,2026,1,200704,12305,310,2157,1,COLEGIO ALEXANDER FLEMING,H1,...,Ciencias Química.,0,0,0,0,Ciencias Técnico Profesional.,0,0,0,0
2,259,2026,2,200704,8005,510,2701,7,LICEO AYTUE,T2,...,Ciencias Química.,0,0,0,0,Ciencias Técnico Profesional.,0,0,0,0
3,496,2026,2,200711,10967,510,60,3,LICEO ELEUTERIO RAMIREZ MOLINA,T2,...,Ciencias Química.,0,0,0,0,Ciencias Técnico Profesional.,0,0,0,0
4,538,2026,2,200711,26538,310,4041,1,COLEGIO TERRAUSTRAL OESTE S.A,H1,...,Ciencias Química.,0,0,0,0,Ciencias Técnico Profesional.,0,0,0,0


Para conocer el detalle de las columnas utilizamos .info(). Dado que el archivo contiene 131 columnas es bastante probable que la funcion retorne un cuadro de resumen basico.

In [32]:
inscritos.info() 

<class 'pandas.DataFrame'>
RangeIndex: 320087 entries, 0 to 320086
Columns: 131 entries, MRUN to OMITIDAS_INV_CTP
dtypes: float64(2), int64(79), str(50)
memory usage: 319.9 MB


Por lo anterior, y como nosotros esperamos obtener información mas precisa, utilizamos la funcion con el argumento: 
verbose=True para forzar mostrar el detalle completo 
show_counts=True para mostrar el conteo de valores no nulos.

In [35]:
inscritos.info(verbose=True) 

<class 'pandas.DataFrame'>
RangeIndex: 320087 entries, 0 to 320086
Data columns (total 131 columns):
 #    Column                   Dtype  
---   ------                   -----  
 0    MRUN                     int64  
 1    ANYO_PROCESO             int64  
 2    COD_SEXO                 int64  
 3    FECHA_NACIMIENTO         int64  
 4    RBD                      str    
 5    CODIGO_ENS               str    
 6    LOCAL_EDUCACIONAL        str    
 7    UNIDAD_EDUCATIVA         str    
 8    NOMBRE_UNIDAD_EDUC       str    
 9    RAMA_EDUCACIONAL         str    
 10   DEPENDENCIA              str    
 11   CODIGO_REGION_EGRESO     str    
 12   NOMBRE_REGION_EGRESO     str    
 13   CODIGO_PROVINCIA_EGRESO  str    
 14   NOMBRE_PROVINCIA_EGRESO  str    
 15   CODIGO_COMUNA_EGRESO     str    
 16   NOMBRE_COMUNA_EGRESO     str    
 17   ANYO_DE_EGRESO           str    
 18   PROMEDIO_NOTAS           float64
 19   PTJE_NEM                 int64  
 20   PORC_SUP_NOTAS           int64  
 2

In [36]:
inscritos.info(verbose=True,show_counts=True) 

<class 'pandas.DataFrame'>
RangeIndex: 320087 entries, 0 to 320086
Data columns (total 131 columns):
 #    Column                   Non-Null Count   Dtype  
---   ------                   --------------   -----  
 0    MRUN                     320087 non-null  int64  
 1    ANYO_PROCESO             320087 non-null  int64  
 2    COD_SEXO                 320087 non-null  int64  
 3    FECHA_NACIMIENTO         320087 non-null  int64  
 4    RBD                      320087 non-null  str    
 5    CODIGO_ENS               320087 non-null  str    
 6    LOCAL_EDUCACIONAL        320087 non-null  str    
 7    UNIDAD_EDUCATIVA         320087 non-null  str    
 8    NOMBRE_UNIDAD_EDUC       316887 non-null  str    
 9    RAMA_EDUCACIONAL         319624 non-null  str    
 10   DEPENDENCIA              320087 non-null  str    
 11   CODIGO_REGION_EGRESO     320087 non-null  str    
 12   NOMBRE_REGION_EGRESO     318876 non-null  str    
 13   CODIGO_PROVINCIA_EGRESO  320087 non-null  str    
 14

La elección final de qué alternativa utilizar depende mayormente del tipo de dataset a revisar. 
 Un caso donde `show_counts=True` cobra valor, es por ejemplo en una encuesta, donde podrían existir valores `null`(vacíos) indicando que una persona no contestó una pregunta. 
 En este dataset en particular vemos que algunas columnas como `NOMBRE_UNIDAD_EDUC`, `RAMA_EDUCACIONAL` o `NOMBRE_REGION_EGRESO`, muestran menos valores no nulos respecto de las observaciones totales, lo cual será revisado en profundidad mas adelante. 

Un detalle particular en esta primera mirada es la definición del tipo de dato de `FECHA_NACIMIENTO`. En el archivo PDF se declara como una cadena o string (str) con formato AAAAMM pero en la info entregada por pandas, se declara como int64 (entero de 64 bits con signo), es decir, un numero. Si bien esto no necesariamente es un error de codificación o lectura, este tipo de detalles son relevantes a la hora de utilizar la información. Por lo que, una vez definido el conjunto de variables que efectivamente se utilizará, se deberán realizar las transformaciones necesarias para tratar de forma correcta la información contenida en cada variable. 

# Revisiones BASE

En un df con 131 columnas, hacer una revision estadistica de cada una parece demasiado, considerando aun la gran cantidad de trabajo que hace el codigo. 
Por lo tanto, partiremos con algunos checks que tienen sentido e iremos definiendo a medida que el trabajo avance, la pertinencia de columnas a analizar en el tiempo. 

La primera duda surge con el identificador MRUN que deberia ser unico por estudiante y considerando los 320087 registros, deberiamos tener esa cantidad de estudiantes. 

In [41]:
# Utilizando entonces la funcion .nunique() validaremos si esta hipotesis es correcta. 
inscritos["MRUN"].nunique() # Usamos ["MRUN"] para definir la columna en particular.
# El resultado es consistente. 


320087

In [ ]:
# Uno podria para este mismo objetivo utilizar tambien .duplicated() con .sum() que suma los duplicados. 
inscritos["MRUN"].duplicated().sum()
# Dado que el resultado es 0, podemos afirmar con las dos formas declaradas, que cada fila corresponde a un alumno. 

np.int64(0)

Otro de los valores iniciales del DF es ANYO_PROCESO. Se podría asumir, dado que los datos corresponden al Proceso de Admisión 2026, que en esta fila deberían ser todos 2026. Lo corroboramos con value_counts() que nos entregara la cantidad de registros donde ANYO_PROCESO es 2026. Si hubiera un 2025 o cualquier otro valor, nos mostrara tambien ese resumen. 

In [ ]:
inscritos["ANYO_PROCESO"].value_counts()
# El resultado es consistente, ANYO_PROCESO solo considera 2026

ANYO_PROCESO
2026    320087
Name: count, dtype: int64